src/plugins/intel_cpu/tests/functional/single_layer_tests/classes/activation.cpp

```diff
@@ -83,6 +83,11 @@ void ActivationLayerCPUTest::generate_inputs(const std::vector<ov::Shape>& targe
             in_data.range = range;
             in_data.resolution = resolution;
             tensor = ov::test::utils::create_and_fill_tensor(funcInput.get_element_type(), targetInputStaticShapes[i], in_data);
+            // cover Sign NAN test case
+            if ((activationType == utils::ActivationTypes::Sign) && targetInputStaticShapes[i].size() == 8 &&
```

Comment:
Jsut wondering why `targetInputStaticShapes[i].size() == 8` condition is needed?

Reply:
The initial purpose of adding this condition is to cover NAN test case only for some of current test instances(e.g. [the current one](https://github.com/openvinotoolkit/openvino/blob/956ad0b761063ea35694e6cb4dfb4947ba7a2f2f/src/plugins/intel_cpu/tests/functional/single_layer_tests/classes/activation.cpp#L267)), and left the  others(e.g. [the current case](https://github.com/openvinotoolkit/openvino/blob/956ad0b761063ea35694e6cb4dfb4947ba7a2f2f/src/plugins/intel_cpu/tests/functional/single_layer_tests/classes/activation.cpp#L266)) to cover none NAN case.

Reply:
It creates implicit dependency between shape and input data content. So if the scope of validated shapes is changed there is a possibility to lose Nans validation.
I would propose to assign Nan to 0th element w/o condition on shape or add additional test parameter that defines should input contains Nans or not.

Reply:
As this operator is elementwise processed, it seems that Nan to 0th element w/o condition can also cover both conditions. Shape condition here are removed now.